gnat2goto/driver/goto_utils.ads

```diff
@@ -40,6 +40,12 @@ package GOTO_Utils is
    function Symbol_Expr (Sym : Symbol) return Irep
      with Post => Kind (Symbol_Expr'Result) = I_Symbol_Expr;
 
+   procedure New_Constant_Symbol_Entry (N : Node_Id;
+                                        A_Symbol_Table : in out Symbol_Table)
+     with Pre => Nkind (N) = N_Object_Declaration;
+     --  Adds a dummy entry to the symbol table to register that a
+     --  constant has already been processed.
```

Comment:
What does this mean?

Reply:
Petr didn't add this, this was already there, and I agree that it was weird.

Reply:
I can see that, but if we're changing this already my opinion is

A) Either we figure out what this means and reword it so it's easier to understand
B) If we can't figure out what this means between the 3 of us I'd posit it's clearly not a very useful comment, so I'd rather remove it in that case.

Reply:
Maybe the comment makes more sense in the call-site: we call it when declaring a constant object, that does not have initial expression -- deferred constant declaration. It's completion is somewhere down the tree but we do not want to process is again which is why store it in the symbol table without the value it will be eventually initialised with.

I moved the comment to the call-site.

Reply:
I also renamed the procedure to `New_Valueless_Object_Symbol_Entry`.